### ISSUE TITLE
[BUG] MLflow scoring server should return 4xx HTTP status code for bad request data

### ISSUE THREAD

**Issue by amesar** [mlflow/mlflow#3354]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **MLflow installed from (source or binary)**:
- **MLflow version (run ``mlflow --version``)**:
- **Python version**:
- **npm version, if running the dev UI**:
- **Exact command to reproduce**:

### Describe the problem

The MLflow scoring server returns a 500 HTTP status code when a bad request is submitted. This happens to servers launched as: 
* `mlflow models serve` 
* `mlflow sagemaker run-local`

The response body contains an error message but the HTTP status code is always 500. The status code should be 400 (BAD REQUEST) which indicates a client error. 5xx codes are reserved for server errors.

### Code to reproduce issue
```
curl  http://localhost:5001/invocations  --verbose \
    -H "Content-Type:application/json" \
    -d ' foo '
```
```
 HTTP/1.1 500 INTERNAL SERVER ERROR
```
### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [x ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by dmatrix** [comment]
@amesar makes sense since this is a [client error](https://www.restapitutorial.com/httpstatuscodes.html).

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one has shipped and can be closed.

The scoring server now returns proper 4xx codes for bad request data:

- Invalid payloads raise `MlflowException(error_code=BAD_REQUEST, ...)` in [`mlflow/pyfunc/scoring_server/__init__.py:172-175`](https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/scoring_server/__init__.py#L172-L175).
- `BAD_REQUEST` is mapped to HTTP 400 in [`mlflow/exceptions.py:49`](https://github.com/mlflow/mlflow/blob/master/mlflow/exceptions.py#L49) via `ERROR_CODE_TO_HTTP_STATUS`.
- The error wrapper in [`mlflow/pyfunc/scoring_server/__init__.py:424-428`](https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/scoring_server/__init__.py#L424-L428) catches `MlflowException` and sets `status_code=e.get_http_status_code()` on the response.

So a malformed payload now yields a 4xx, not a 500, which matches the original ask.

Would you mind closing this out? If something specific is still missing — a particular bad-input path that still 5xxs — point me at it and I'll dig further.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `master` branch.)
